### PR TITLE
feat/add meta information consistency attribute

### LINKED ISSUE #300: Container consistency policy
## Is your feature request related to a problem? Please describe.
I'm always frustrated when we can't specify consistency policy for a container.

## Describe the solution you'd like
Add an optional enum field with the following options:
 * available (just like the old one, no meta)
 * strict (nodes use meta and wait for meta acceptance)
 * optimistic (nodes use meta, but do not wait for acceptance)

## Describe alternatives you've considered
If we want to alternate between different modes and preserve compatibility, there are not a lot of them.

### PR DESCRIPTION
**Pull request by carpawell**
Closes #300.

### CODE REVIEW

**Comment by carpawell** [comment]
Naming is from the issue, have no suggestion from my side. I do not like that attribute idea, cause it is more like a core feature for me, every container gonna have it and it changes the game significantly, but agree that an attribute thing makes it more flexible.

**Comment by carpawell** [comment]
Comments may be adjusted in the future when we will have more understanding of what the meta flow looks like.

File: container/types.proto [hunk 1]
```diff
@@ -52,6 +52,22 @@ message Container {
   //   accepted in a NeoFS network only if the global network hashing configuration
   //   value corresponds with that attribute's value. After container inclusion, network
   //   setting is ignored.
+  // * __NEOFS__METAINFO_CONSISTENCY \
+  //   Policy rule that defines the condition under which an object is considered
+  //   processed. Acceptable values and meanings:
+  //     1. <empty>/<missing attribute>: SN does not process objects' meta
```
Comment (cthulhu-rider) [inline container/types.proto:58]: attr value cannot be empty as stated above, absence is enough

File: container/types.proto [hunk 2]
```diff
@@ -52,6 +52,22 @@ message Container {
...
+  //       information, it is not indexed and object presence/number of copies
+  //       is not proved after a successful object PUT operation; the behavior
+  //       is the same as it was before this attribute introduction
+  //     2. "strict": SN processes objects' meta information, it is validated,
+  //       indexed and signed accordingly by a required minimal number of nodes
+  //       that are included in the container, a corresponding object inclusion
+  //       notification can be catch
```
Comment (cthulhu-rider) [inline container/types.proto:65]:
```suggestion
  //       notification can be caught
```

File: container/types.proto [hunk 3]
```diff
@@ -52,6 +52,22 @@ message Container {
   //   accepted in a NeoFS network only if the global network hashing configuration
   //   value corresponds with that attribute's value. After container inclusion, network
   //   setting is ignored.
+  // * __NEOFS__METAINFO_CONSISTENCY \
+  //   Policy rule that defines the condition under which an object is considered
+  //   processed. Acceptable values and meanings:
+  //     1. <empty>/<missing attribute>: SN does not process objects' meta
+  //       information, it is not indexed and object presence/number of copies
+  //       is not proved after a successful object PUT operation; the behavior
```
Comment (roman-khimov) [inline container/types.proto:60]: proven?

File: container/types.proto (hunk 1 above)
Comment (carpawell) [reply container/types.proto:58]: ok, reworded it to sound like "other cases" with the old behavior

File: container/types.proto (hunk 3 above)
Comment (carpawell) [reply container/types.proto:60]: afaik, both ways are ok currently, but don't mind

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**